### DISCUSSION TITLE
blocklist.txt place

### DISCUSSION THREAD

**Discussion by squromiv** [TechnitiumSoftware/DnsServer#586] [Q&A] [answered]
I run Technitium DNSServer by "docker-compose up -d" as described [here](https://blog.technitium.com/2017/11/running-dns-server-on-ubuntu-linux.html). So where  blocklist.txt should be exactly placed in order to edit it and to set as custom block list?

P.S. In Windows it is situated in "c:\Program Files (x86)\Technitium\DNS Server\www\hosts\". But with Ubuntu I am a bit confused and did not find.

**Comment by ShreyasZare** [comment, marked as answer]
> I run Technitium DNSServer by "docker-compose up -d" as described [here](https://blog.technitium.com/2017/11/running-dns-server-on-ubuntu-linux.html). So where blocklist.txt should be exactly placed in order to edit it and to set as custom block list?

You can place it in the "www" folder of the DNS web panel and use the `http://localhost:5380/blocklist.txt` as the URL.

> P.S. In Windows it is situated in "c:\Program Files (x86)\Technitium\DNS Server\www\hosts". But with Ubuntu I am a bit confused and did not find.

There is no such "hosts" folder unless you have created it.  If you create such a folder and place "blocklist.txt" in it then your URL will be `http://localhost:5380/hosts/blocklist.txt`. You will find the "www" folder in `/etc/dns/www` on Linux but since you are using docker, you will need to create a volume first for that path and then place your txt file in it on the host folder you mount the volume on.

**Reply by squromiv** [reply]
> you will need to create a volume first for that path and then place your txt file in it on the host folder you mount the volume on.

Is it possible somehow to automate this process with docker-compose.yaml file etc? I am a real newbie in Linux.

**Reply by ShreyasZare** [reply]
Just define another volume for path `/etc/dns/www` in your docker compose file which will map to a folder on your host OS. Then create the container and put your file in that folder.

If you fix this complex then you can host the blocklist.txt on any other web server that you have running on your setup and use the URL for that webserver in the DNS server block list settings.
